eland/operations.py

```diff
@@ -1250,6 +1250,40 @@ def to_csv(  # type: ignore
         if path_or_buf is None:
             return "".join(result)
 
+    def to_json(  # type: ignore
+        self,
+        query_compiler: "QueryCompiler",
+        show_progress=False,
+        path_or_buf=None,
+        orient=None,
+        lines=False,
+        **kwargs,
+    ):
+        if orient == "records" and lines is True and isinstance(path_or_buf, str):
+            result = []
+            processed = 0
+            with open(path_or_buf, "w") as w:
+                for i, df in enumerate(
+                    self.search_yield_pandas_dataframes(query_compiler=query_compiler)
+                ):
+                    if (
+                        show_progress
+                        and processed % DEFAULT_PROGRESS_REPORTING_NUM_ROWS == 0
+                    ):
+                        print(f"{datetime.now()}: read {processed} rows")
+                    output = df.to_json(
+                        orient=orient,
+                        lines=lines,
+                        **kwargs,
+                    )
+                    w.write(output.strip())
+                    w.write("\n")
+            return "".join(result) or None
```

Comment:
What is the point of this? result is always the empty array

Reply:
Good point, this will contain something now.